Formula/fn.rb

```diff
@@ -0,0 +1,24 @@
+class Fn < Formula
+  desc "CLI tool for the fn project."
+  homepage "https://fnproject.github.io"
+  url "https://github.com/fnproject/cli/archive/0.3.72.tar.gz"
+  sha256 "42f19a50cfb649fa2677ae1e1bb09847e9eb533a75658446e20d84afb086c71b"
+
+  depends_on "go" => :build
+  depends_on "glide" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GLIDE_HOME"] = HOMEBREW_CACHE/"glide_home/#{name}"
+    dir = buildpath/"src/github.com/fnproject/cli"
+    dir.install Dir["*"]
+    cd dir do
+      system "glide", "install", "-v", "--force", "--skip-test"
+      system "go", "build", "-o", "#{bin}/fn"
```

Comment:
Haven't checked but I'd bet a dollar you need a `prefix.install_metafiles` inside this block.

Reply:
```
.
└── 0.3.72
    ├── INSTALL_RECEIPT.json
    └── bin
        └── fn

2 directories, 2 files
```
and a :dollar: goes to @DomT4

Reply:
It still bothers me we've never found a completely sensible way to automate checking this for `go`-using formulae, which are by far the most common offenders :see_no_evil:.

Reply:
Just added that. What is that for exactly?  I can't seem to find any docs on it, it's not in the Formula Cookbook or anything.

Reply:
Yeah, IMO it not being massively documented isn't incorrect on brew's part, because the existing behaviour here should be classed as a minor bug. Metafiles are things like the README, LICENSE, CHANGELOG, etc. There's a full list of what brew looks for [here](https://github.com/Homebrew/brew/blob/79e8cdd3ed8c7f5dee573648d55eb3d3e305de19/Library/Homebrew/metafiles.rb#L5-L12).

Normally brew does a really decent job of finding & installing those files automatically & the detection is pretty solid, but things start getting a bit wobbly once you transpose the contents of the `buildpath` to another directory, as is especially common with `go`-using formulae due to Go's pedantry around tree layout.